### ISSUE TITLE
Support unique constraints on tables without a primary key

### ISSUE THREAD

**Issue by latot** [dolthub/dolt#3165]
Hi hi, here again testing spatial data, testing water from the OSM, seems unique constrains is still not supported:

```
ogr2ogr 'MySQL:water,user=root,password=,host=127.0.0.1,port=3300' gis_osm_water_a_free_1.shp
ERROR 1: MySQL error message:unsupported feature: unique constraint on keyless tables Description: CREATE TABLE `gis_osm_water_a_free_1` (    OGR_FID INT UNIQUE NOT NULL AUTO_INCREMENT,    SHAPE GEOMETRY NOT NULL)
ERROR 1: Terminating translation prematurely after failed
translation of layer gis_osm_water_a_free_1 (use -skipfailures to skip errors)
```

```
Starting server with Config HP="localhost:3300"|U="root"|P=""|T="28800000"|R="false"|L="info"
2022-04-06T12:30:46-04:00 INFO [conn 4] NewConnection {DisableClientMultiStatements=false}
2022-04-06T12:30:46-04:00 WARN [conn 4] error running query {connectTime=2022-04-06T12:30:46-04:00, error=unsupported feature: unique constraint on keyless tables, query=CREATE TABLE `gis_osm_water_a_free_1` ( OGR_FID INT UNIQUE NOT NULL AUTO_INCREMENT, SHAPE GEOMETRY NOT NULL)}
2022-04-06T12:30:46-04:00 INFO [conn 4] ConnectionClosed {}
```

Thx.

**Comment by zachmu** [comment]
Thanks for the report!

This should be an easy fix, just an oversight on our part (indexes on keyless tables are already supported).

**Comment by VinaiRachakonda** [comment]
Addressed in #3245
